x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -295,6 +296,37 @@ public List<Setting<?>> getSettings() {
         return List.of(NUM_FAILURE_RETRIES_SETTING, SCHEDULER_FREQUENCY);
     }
 
+    @Override
+    public Settings additionalSettings() {
+
+        String transformConfigVersionAttrName = "node.attr." + TransformConfigVersion.TRANSFORM_CONFIG_VERSION_NODE_ATTR;
+
+        Settings.Builder additionalSettings = Settings.builder();
+
+        addTransformNodeAttribute(additionalSettings, transformConfigVersionAttrName, TransformConfigVersion.CURRENT.toString());
+        return additionalSettings.build();
+    }
+
+    private void addTransformNodeAttribute(Settings.Builder additionalSettings, String attrName, String value) {
+        String oldValue = settings.get(attrName);
+        if (oldValue == null) {
+            additionalSettings.put(attrName, value);
+        } else {
+            reportClashingNodeAttribute(attrName);
+        }
+    }
+
+    private void reportClashingNodeAttribute(String attrName) {
+        throw new IllegalArgumentException(
+            "Directly setting ["
+                + attrName
+                + "] is not permitted - "
+                + "it is reserved for the transform plugin. If your intention was to customize the transform plugin, set the ["
+                + attrName.replace("node.attr.", "xpack.")
+                + "] setting instead."
```

Comment:
The last sentence of this error isn't true for this particular node attribute, so please remove it.

(This means there's another bug in the ML plugin. The second sentence there will apply to some but not all of the attributes.)